lib/expressions.js

```diff
@@ -71,10 +76,70 @@ class OperationExpression {
   }
 
   toJavaScript() {
-    const parameters = Array.from(this.identifiers);
-    const header = '(' + parameters.join(', ') + ')';
-    return header + ' => ' + this.toExpression();
+    const header = `(${this.identifiers.join(', ')})`;
+    return `${header} => ${this.toExpression()}`;
+  }
+}
+
+class ConditionExpression {
+  #identifiers = new Set();
+
+  constructor(clauses) {
+    this.clauses = clauses;
+
+    for (const { condition, consequents } of clauses) {
+      const conditionIdentifiers = condition.identifiers ?? [];
+      const consequentIdentifiers = consequents.flatMap(
+        (consequent) => consequent.identifiers ?? [],
+      );
+
+      for (const name of [...conditionIdentifiers, ...consequentIdentifiers]) {
+        this.#identifiers.add(name);
+      }
+    }
+  }
+
+  get identifiers() {
+    return Array.from(this.#identifiers);
+  }
+
+  // eslint-disable-next-line consistent-return
```

Comment:
Lisp returns `nil` if there are no matches. What behavior do we expect? Currently it returns `undefined` in this case